spec/auth.md

```diff
@@ -0,0 +1,122 @@
+
+The following section explains how to acquire such a token.
+
+## Token Request API Endpoints
+
+**URL Pattern:**
+
+```txt
+https://huggingface.co/api/{repo_type}s/{repo_id}/xet-{token_type}-token/{revision}
+```
+
+**Parameters:**
+
+- `repo_type`: Type of repository - `model`, `dataset`, or `space`
+- `repo_id`: Repository identifier in format `namespace/repo-name`
+- `token_type`: Either `read` or `write`.
+- `revision`: Git revision (branch, tag, or commit hash; default to using `main`)
+
+To understand the distinction for between `token_type` values read onwards in this document to [Token Scope](../spec/auth.md#token-scope).
+
+**Example URLs:**
+
+```txt
+https://huggingface.co/api/models/sentence-transformers/all-MiniLM-L6-v2/xet-read-token/main
+https://huggingface.co/api/datasets/HuggingFaceM4/the_cauldron/xet-write-token/v1.1
+https://huggingface.co/api/spaces/jsulz/ready-xet-go/xet-read-token/main
+```
+
+**HTTP Method:** GET
+
+**Required Headers:**
+
+- `Authorization`: Bearer token for Huggingface Hub authentication
+
+### Response Format
+
+A JSON encoded object with the following format:
+
+```typescript
+{
+   "accessToken": string,
+   "exp": number,
+   "casUrl": string,
+}
+```
+
+- accessToken is the token to be used when invoking API's on the Xet CAS service (any Xet API denoted in this specification)
+- exp is the unix timestamp of when this token expires
+- casUrl is the API service endpoint URL
+
+#### Example Response Object
+
+```json
+{
+   "accessToken": "xet_xxxxxxxxxxx",
+   "exp": 1848535668,
+   "casUrl": "https://cas-server.xethub.hf.co"
+}
+```
+
+### Error Handling
+
+#### HTTP Errors
+
+- **401 Unauthorized**: Invalid or missing Hub authentication token
+- **403 Forbidden**: Insufficient permissions for the requested token type
+- **404 Not Found**: Repository or revision does not exist
+
+### Implementation Example
+
+Here's a basic implementation flow:
+
+1. **Make the request:**
+
+   ```http
+   GET /api/models/black-forest-labs/FLUX.1-dev/xet-read-token/main
+   Host: huggingface.co
+   Authorization: Bearer xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+   ```
+
+2. **Parse the response**
+
+   ```python
+   endpoint = response_json["casUrl"]
+   access_token = response_json["accessToken"]
+   expiration = response_json["exp"]
+   ```
+
+3. **Use the token with Xet service:**
+   Use Bearer authentication with the `access_token` to authenticate with the Xet service at `endpoint` until `expiration` time.
+
+4. **Token refresh (when needed):**
+   Use the same API to generate a new token
+
+## Token Scope
+
+Xet tokens can have either a `read` or a `write` scope. The type of token issues is determined on the `token_type` URI path component when requesting the token from the HuggingFace Hub (see above).
+
+Revise API specification for what scope level is required to invoke each API (briefly, only `POST /shard` and `POST /xorb/*` API's require `write` scope).
```

Comment:
It would be helpful to mention that `write` scope allows for `read` on a resource.